qemu/qemu.go

```diff
@@ -1345,6 +1370,42 @@ func (config *Config) appendMemory() {
 
 		config.qemuParams = append(config.qemuParams, "-m")
 		config.qemuParams = append(config.qemuParams, strings.Join(memoryParams, ""))
+
+		hugePages := config.Memory.HugePages || config.Knobs.HugePages
+		memPrealloc := config.Memory.MemPrealloc || config.Knobs.MemPrealloc || hugePages
+		share := config.Memory.Share || hugePages
```

Comment:
do huge pages enable memory pralloc and share in QEMU?  ummm I don't think so

Reply:
Original code (before this pr) enable share and prealloc, this pr keeps the behavior.